extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -367,7 +367,7 @@ public void start()
         exec.submit(
             () -> {
               try {
-                while (!Thread.currentThread().isInterrupted()) {
+                while (!Thread.currentThread().isInterrupted() && !stopped) {
```

Comment:
In addition to this, I suggest adding a timeout to `take()`. Otherwise this thread might block on `take()` forever and never actually get to check `!stopped`. If `take()` times out, we should check `stopped`, then if it's still false, run `take()` again (again, with a timeout).

Reply:
Make sense,  I'll do that.